packages/phone-number-privacy/combiner/src/match-making/get-contact-matches.ts

```diff
@@ -36,16 +42,37 @@ export async function handleGetContactMatches(
       return
     }
 
-    const { account, userPhoneNumber, contactPhoneNumbers, hashedPhoneNumber } = request.body
+    const {
+      account,
+      userPhoneNumber,
+      contactPhoneNumbers,
+      hashedPhoneNumber,
+      blindedQueryPhoneNumber,
+    } = request.body
 
     if (!(await isVerified(account, hashedPhoneNumber, getContractKit(), logger))) {
       respondWithError(response, 403, WarningMessage.UNVERIFIED_USER_ATTEMPT_TO_MATCHMAKE, logger)
       return
     }
+
     if (await getDidMatchmaking(account, logger)) {
-      respondWithError(response, 403, WarningMessage.DUPLICATE_REQUEST_TO_MATCHMAKE, logger)
-      return
+      const blindedPhoneNumberRecord = await getAccountBlindedPhoneNumber(account, logger)
```

Comment:
Thinking more about using the blindedPhoneNumber.. I'm not sure it'll work. We can't tell that the `blindedQueryPhoneNumber` maps to the `hashedPhoneNumber` right? If I wanted to perform many unique queries, I could keep the `blindedQueryPhoneNumber` the same while changing the `hashedPhoneNumber`. Perhaps instead, we can store in the DB a hash of (`hashedPhoneNumber` + `account`) which will be unique per pair.

Reply:
Yea I was thinking that too. Using a hash of (hashedPhoneNumber + account) would be a little better but someone could still use a rainbow table to find the phone number for a specific account. What we really want is the following 

Client                                                                        --->  combiner 
(1) blind(dekPrivateKey, hashedPhoneNumber)            (2) verify(dekPublicKey, hashedPhoneNumber, blindedMessage)
(3) unblind(dekPrivateKey, blindedMessage)       <---

Reply:
Where step (2) checks that the blinded message was blinded by the DEK without knowing the DEK private key. We may need some help from the crypto circle if we decided to do this.